### DISCUSSION TITLE
Info:  Why is ax::Color4B faster as ax::Color4F?

### DISCUSSION THREAD

**Discussion by aismann** [axmolengine/axmol#2193] [Q&A] [answered]
The background:
On class DrawNode the Color4F parameter was changed to Color4B.
```
See also axmol-1.0.0 Aug.9 2023:
[REFINE] Change DrawNode api color parameters from Color4F to low Color4B
```
Thats what I have found on the WWW: => Its seem Color4F is faster?

`Colors in computer graphics are represented as an array of 4 values: the red, green, blue and alpha (opacity) component, commonly abbreviated to RGBA. When defining a color in OpenGL or GLSL we set the strength of each component to a value between 0.0 and 1.0. If, for example, we would set red to 1.0 and green to 1.0 we would get a mixture of both colors and get the color yellow. Given those 3 color components we can generate over 16 million different colors!`

Thanks for clarify.

**Comment by halx99** [comment, marked as answer]
The axmol/cocos2d-x use Color4B transfer color data to GPU in vertex format of builtin shaders

**Reply by aismann** [reply]
Thanks for this fast answer. I will made a Wiki entry too.
I updated the title also.

**Comment by aismann** [comment]
with axmol 3.0 it can closed

**Comment by aismann** [comment]
.
